selfdrive/car/gm/carstate.py

```diff
@@ -107,6 +108,7 @@ def update(self, pt_cp, cam_cp, loopback_cp):
                       pt_cp.vl["EBCMFrictionBrakeStatus"]["FrictionBrakeUnavailable"] == 1)
 
     ret.cruiseState.enabled = pt_cp.vl["AcceleratorPedal2"]["CruiseState"] != AccState.OFF
+    self.cruise_resume_allowed = self.cruise_resume_allowed or ret.cruiseState.enabled
```

Comment:
You also need to consider the main switch https://github.com/commaai/openpilot/pull/908